src/config/ConfigManager.hpp

```diff
@@ -7,9 +7,11 @@
 #include <vector>
 #include <memory>
 
+std::string getMainConfigPath();
```

Comment:
uhh, no. Make it static local in the configManager.cpp file and use it if configPath is empty.

Reply:
I changed it to a ternary, tell me (or feel free to edit it directly) if you have a preference to something else.